tests/jenkins/TestDockerBuild.groovy

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+import jenkins.tests.BuildPipelineTest
+import org.junit.Before
+import org.junit.Test
+import static com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.library
+import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
+import static org.assertj.core.api.Assertions.assertThat
+
+class TestDockerBuild extends BuildPipelineTest {
+
+    String dockerBuildGitRespository = 'https://github.com/opensearch-project/opensearch-build'
+    String dockerBuildGitRespositoryReference = 'main'
+    String dockerBuildScriptwithCommands = 'bash docker/ci/build-image-multi-arch.sh -v <TAG_NAME> -f <DOCKERFILE PATH>'
+
+    @Override
+    @Before
+    void setUp() {
+
+        super.setUp()
+
+        // Variables
+        binding.setVariable('DOCKER_BUILD_GIT_REPOSITORY', dockerBuildGitRespository)
+        binding.setVariable('DOCKER_BUILD_GIT_REPOSITORY_REFERENCE', dockerBuildGitRespositoryReference)
+        binding.setVariable('DOCKER_BUILD_SCRIPT_WITH_COMMANDS', dockerBuildScriptwithCommands)
+
+    }
+
+    @Test
+    void DockerBuildRegression() {
+        super.testPipeline('jenkins/docker/docker-build.jenkinsfile',
+                'tests/jenkins/jenkinsjob-regression-files/docker/docker-build.jenkinsfile')
+    }
+
+    @Test
```

Comment:
I believe we also need to validate:
* docker-build.sh is called. https://github.com/opensearch-project/opensearch-build/pull/2790/files#diff-0d66138b0d41c0c6753e22a4934493a3b56d5c5f303f0d047e9fda4aa4cb7745R13-R16
* Right credential id is used as change in creds will result in failure

So 2 more tests.

Reply:
Sounds good, @gaiksaya .. let me add test case to make sure the 'docker-build.sh' is called and successful. 

As for the crendential Id .. If i got it right, isn't it part of ENV set-up on the CI? Will it possibly be changed by user who created new jenkinsfile?

Reply:
So if someone makes a change to this jenkinsfile (changed the credential id) it won't be caught until we run this file right? So better way to avoid that is add a test saying this is the exact credential we want to use for this file